### PR TITLE
reduce surprise in choice of CASE/String/STRING by doing case-independent comparisons

### PR DESCRIPTION
**Pull request by mcr**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

Where should this change be documented?
possibilities: https://www.openssl.org/docs/manmaster/man5/x509v3_config.html https://www.openssl.org/docs/manmaster/man3/ASN1_generate_nconf.html

### CODE REVIEW

File: crypto/asn1/asn1_gen.c [hunk 1]
```diff
@@ -564,7 +564,7 @@ static int asn1_str2tag(const char *tagstr, int len)
 
     tntmp = tnst;
     for (i = 0; i < OSSL_NELEM(tnst); i++, tntmp++) {
-        if ((len == tntmp->len) && (strncmp(tntmp->strnam, tagstr, len) == 0))
```
Comment (slontis) [inline crypto/asn1/asn1_gen.c:567]: windows build failure is relevant..

#include "e_os.h"                /* strncasecmp() */

**Comment by slontis** [comment]
Maybe add your description as to the why this is needed into the commit message?

**Comment by richsalz** [comment]
Document it like this:
```
diff --git a/doc/man3/ASN1_generate_nconf.pod b/doc/man3/ASN1_generate_nconf.pod
index bf22e3624c..256ed9f99c 100644
--- a/doc/man3/ASN1_generate_nconf.pod
+++ b/doc/man3/ASN1_generate_nconf.pod
@@ -42,8 +42,9 @@ I<value> and I<modifier> are explained below.
 
 =head2 Supported Types
 
-The supported types are listed below. Unless otherwise specified
-only the B<ASCII> format is permissible.
+The supported types are listed below.
+Case is not significant in the type names.
+Unless otherwise specified only the B<ASCII> format is permissible.

```

**Comment by richsalz** [comment]
For a test, take one of the existing "cnf" files and change the capitalization, perhaps the ca-and-certificate.cnf file.

**Review by paulidale — APPROVED** [review]

**Comment by mcr** [comment]
> ca-and-certificate.cnf
I looked through the test/*.cnf, include ca-and-certs.cnf, and I don't see any attributes in these files that would apply.
Most of the .c tests use the #define for these attributes.
I grep'ed through the tests to see if I could find an obvious place where I could change the case, but I didn't find something.

**Comment by richsalz** [comment]
So replace one of the defines with a mixed-case identifier.  :)

**Comment by mcr** [comment]
> So replace one of the defines with a mixed-case identifier. :)

In the CNF files, there are no cases where the relevant strings are used.
In the C code, they are #define, which are not strings that get evaluated. Changing the defines into a mixed-case will just result in a compiler failure.

**Comment by richsalz** [comment]
I meant the *value* of a define.  But I guess there are no defines that have ASN1 in them.  Oh well.

**Review by t8m — APPROVED** [review]

**Review by slontis — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master, thanks for the contribution.

**Pull request closed without merging**